vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/data/SelectDataView.java

```diff
@@ -0,0 +1,51 @@
+package com.vaadin.flow.component.select.data;
+
+import java.util.Optional;
+
+import com.vaadin.flow.data.provider.DataView;
+
+/**
+ * An extended {@link DataView} interface for Select component
+ * 
+ * @param <T>
+ *            item type
+ */
+public interface SelectDataView<T> extends DataView<T> {
+    /**
+     * Gets the item at the given index in the data set.
+     *
+     * @param index
+     *            index to get item at
+     * @return item at index
+     * @throws IndexOutOfBoundsException
+     *             requested index is outside of the available data set range.
+     */
+    T getItemOnIndex(int index);
+
+    /**
+     * Selects the item at the given index in the data set.
```

Comment:
So in case there is a filter applied, does it matter here ? What about sorting ? Answer in javadocs

Reply:
done